### ISSUE TITLE
Power Builder not working

### ISSUE THREAD

**Issue by Adunakhor** [templatemonster/power-builder-upd#19]
Logged into a customer's site today that is only using Power Builder on their home page. Power Builder is listed as the default editor for the home page, but will  not open. It is activated and is on version 1.3.1. Power Builder Integrator is on version 1.0.12. The site is currently on WordPress version 5.7.2. Not sure when Power Builder stopped working as I had not logged into that site in more than a year to make edits. I saw folks on here saying that they had to uninstall and reinstall an older version of Power Builder to get it to work after WordPress 5.0 update (or higher), but I never upgraded Power Builder above 1.3.1, which was their suggested working version.

I need to make edits to the home page on this site, but I am unable to do so without Power Builder working. Any suggestions?

BTW, I have always had trouble with PB on this site, because it often lost settings for blocks after I saved them. I decided that I would never use this plugin again, and this issue only proves that was a smart decision.

**Comment by fabriziosorbara** [comment]
The problem is in the "cherry testimonial" plugin 

\'ready\' event is deprecated and it is in wp-content/plugins/cherry-testi/cherry-framework/modules/cherry-js-core/assets/js/min/cherry-js-core.min.js

the code (unminified is)

var CherryJsCore = {};
!(function (e) {
    "use strict";
    (CherryJsCore = {
        name: "Cherry Js Core",
        version: "1.0.0",
        author: "Cherry Team",
        variable: {
            $document: e(document),
            $window: e(window),
            browser_supported: !0,
            security: window.cherry_ajax,
            loaded_assets: { script: window.wp_load_script, style: window.wp_load_style },
            ui_auto_init: "true" === window.ui_init_object.auto_init,
            ui_auto_target: window.ui_init_object.targets,
        },
        status: { on_load: !1, is_ready: !1 },
        init: function () {
            e(document).on("ready", CherryJsCore.ready), e(window).on("load", CherryJsCore.load);
        },
        ready: function () {
            (CherryJsCore.status.is_ready = !0), CherryJsCore.expressions.widget_ui_init();
        },
        load: function () {
            CherryJsCore.status.on_load = !0;
        },
        expressions: {
            widget_ui_init: function () {
                e(document).on("widget-added widget-updated", function (r, t) {
                    e("body").trigger({ type: "cherry-ui-elements-init", _target: t });
                });
            },
        },
        utilites: {
            namespace: function (e) {
                var r = e.split("."),
                    t = CherryJsCore,
                    i = r.length,
                    o = 0;
                for (o = 0; o < i; o += 1) void 0 === t[r[o]] && (t[r[o]] = {}), (t = t[r[o]]);
                return t;
            },
        },
    }).init();
})(jQuery);




COULD someone try to update it?

**Comment by Adunakhor** [comment]
Thanks. I will give that a try. Much appreciated.

 

Thanks,

 

Shane

**Comment by Adunakhor** [comment]
Deactivating the Cherry Testimonial Plugin has not affected my ability to use the Power Builder tool. 

 

The problem may simply be that Cherry has not offered an update to its framework since version 5.

 

If I simply edit the .js what would I need in lieu of the ready statement? What would be the correct code?

 

Any help you can provide would be beneficial to keeping Power Builder on this site.

 

Thanks,

 

Shane

**Comment by fabriziosorbara** [comment]
You're right; i did not look at the whole problem, there are still others deprecations in the whole cherry framework :( at least 24....

**Comment by Adunakhor** [comment]
Thanks for looking at it again.

 

Is there no other way to activate Power Builder on this site? I just want to access and edit that home page in Power Builder, but it looks like I may just need to rebuild it in a different tool.

 

Thanks,

 

Shane

**Comment by ricketstar** [comment]
Experiencing an issue as well - editor won't load at all. This means a whole lot of work to make small modifications.

**Comment by thetgc** [comment]
I have posted this in another comment but I think it is worth repeating here.

There is a workaround for this but you need to roll back to WordPress version 5.5. Power Builder ceases to function past that version of WordPress. I downloaded a plugin called WP Downgrade. Once installed go into your plugins folder and click on its settings. You can then choose a version of WP to roll back to. Then complete your edits to your site and move back to the latest version of WP. It worked well for me on a site I built in 2017.
